Model/Order/PlaceOrder.php

```diff
@@ -261,158 +256,41 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
             return $this->responseFactory->create(
                 [
                     'errors' => [
                         $this->errorFactory->create(
                             [
-                                'message' => __(
-                                    'There are no authorized payments available for order "%1"',
-                                    $publicOrderId
-                                ),
-                                'code' => 500,
-                                'type' => 'server.bold_checkout_api_error'
+                                'message' => $authorizedPayments['error'],
+                                'code' => 422,
+                                'type' => $errorType
                             ]
                         )
                     ]
                 ]
             );
         }
 
-        $transactions = array_filter(
-            $authorizedPayments['data']['transactions'],
-            /**
-             * @param array{
-             *     gateway: string,
-             *     payment_id: string,
-             *     amount: int,
-             *     transaction_id: string,
-             *     currency: string,
-             *     step: string,
-             *     status: 'success'|'failed'|'',
-             *     tender_type: string,
-             *     tender_details: array{
-             *         brand: string,
-             *         last_four: string,
-             *         bin: string,
-             *         expiration: string
-             *     },
-             *     gateway_response_data: string[]
-             * } $transaction
-             */
-            static function (array $transaction): bool {
-                return $transaction['status'] !== 'failed';
-            }
-        );
-
-        if (count($transactions) === 0) {
+        $firstTransaction = $authorizedPayments['data']['transactions'][0] ?? [];
+        if (empty($firstTransaction) || $firstTransaction['status'] === 'failed') {
             return $this->responseFactory->create(
                 [
                     'errors' => [
                         $this->errorFactory->create(
                             [
-                                'message' => __(
-                                    'There are no successful transactions available for order "%1"',
-                                    $publicOrderId
-                                ),
-                                'code' => 500,
-                                'type' => 'server.bold_checkout_api_error'
+                                'message' => 'Invalid Transaction',
+                                'code' => 422,
+                                'type' => 'server.payment_authorization_error'
                             ]
                         )
                     ]
                 ]
             );
         }
 
-        /**
-         * @var array{
```

Comment:
[Preference] I removed these doc types for two reasons: 
1. As the methods change it's easy to not update these and they can quickly become out of date 
2. They're not really used in most of the module. 
Let me know if there's any strong preferences here and happy to change back.

Reply:
I added them for doing static analysis with [PHPStan](https://phpstan.org). They also provide a visual reference for what values are expected to be in the array, and I think that PhpStorm uses them to provide value hints.